lib/pat.c

```diff
@@ -92,6 +93,12 @@ typedef struct {
 #define PAT_DELETING  (1 << 1)
 #define PAT_IMMEDIATE (1 << 2)
 
+static inline bool
+pat_is_key_large(grn_pat *pat)
+{
+  return (pat->header->flags & GRN_OBJ_KEY_LARGE) == GRN_OBJ_KEY_LARGE;
```

Comment:
Could we write the following? 
```suggestion
  return pat->header->flags & GRN_OBJ_KEY_LARGE;
```

Reply:
📝 The existed code in `hash.c` uses the same approach now here.
https://github.com/groonga/groonga/blob/58a20e712ac4178bcb13d1792c24701cd29d97c0/lib/hash.c#L5160-L5164

Reply:
The implement of this function like `grn_hash_is_large_total_key_size()`(https://github.com/groonga/groonga/blob/v15.1.1/lib/hash.c#L5161).

Also, the return value type of this function is bool.
However, the result of `pat->header->flags & GRN_OBJ_KEY_LARGE` is `0` or `4096`.
Probably, the above value is typecasted implicit, but I think that we should not except behavior implicit.